packages/grid/src/utils/types.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+export type TYPE_ALIGN_ITEMS = 'start' | 'end' | 'center' | 'baseline' | 'stretch';
+export const ARRAY_ALIGN_ITEMS: Array<TYPE_ALIGN_ITEMS> = [
+  'start',
+  'end',
+  'center',
+  'baseline',
+  'stretch'
+];
+
+export type TYPE_ALIGN_SELF = 'auto' | TYPE_ALIGN_ITEMS;
+export const ARRAY_ALIGN_SELF: Array<TYPE_ALIGN_SELF> = [
+  'auto',
+  'start',
+  'end',
+  'center',
+  'baseline',
+  'stretch'
+];
+
+export type TYPE_DIRECTION = 'row' | 'row-reverse' | 'column' | 'column-reverse';
+export const ARRAY_DIRECTION: Array<TYPE_DIRECTION> = [
+  'row',
+  'row-reverse',
+  'column',
+  'column-reverse'
+];
+
+export type TYPE_JUSTIFY_CONTENT = 'start' | 'end' | 'center' | 'between' | 'around';
+export const ARRAY_JUSTIFY_CONTENT: Array<TYPE_JUSTIFY_CONTENT> = [
+  'start',
+  'end',
+  'center',
+  'between',
+  'around'
+];
+
+export type TYPE_NUMBER = number | string;
```

Comment:
This naming seems ambiguous right now. `foo = foo | bar`

Is there a more specific name for this type that we could use? i.e. `COLUMN_SIZE`

Reply:
I had it particularly named originally. But as you go through the diff, you'll see that it applies anywhere that a prop accepts a numeric value – not just columns. The generalized naming works better.

Reply:
> it applies anywhere that a prop accepts a numeric value – not just columns

Then let's name this `GRID_NUMBER` or something generic. Normally with something as basic as `number | string` you would just add that to the other type definitions.